sycl/test/basic_tests/builtins_implicitly_convertible_args.cpp

```diff
@@ -0,0 +1,172 @@
+// RUN: %clangxx -fsycl -fpreview-breaking-changes -fsyntax-only %s -o %t
+// REQUIRES: preview-breaking-changes-supported
+
+// Checks that builtins without template arguments allow for implicit
+// conversions of arguments.
+
+#include <sycl/sycl.hpp>
+
+template <typename T> struct ImplicitlyConvertibleType {
+  operator T() const { return {}; }
+};
+
+#define UNARY_CHECK_INNER(FUNC_NAME, ...)                                      \
+  static_assert(std::is_same_v<                                                \
+                decltype(sycl::FUNC_NAME(                                      \
+                    std::declval<ImplicitlyConvertibleType<__VA_ARGS__>>())),  \
+                decltype(sycl::FUNC_NAME(std::declval<__VA_ARGS__>()))>);
+
+#define FLOAT_UNARY_CHECK(FUNC_NAME) UNARY_CHECK_INNER(FUNC_NAME, float)
+
+#define GENFLOAT_UNARY_CHECK(FUNC_NAME)                                        \
+  FLOAT_UNARY_CHECK(FUNC_NAME)                                                 \
+  UNARY_CHECK_INNER(FUNC_NAME, sycl::half)                                     \
+  UNARY_CHECK_INNER(FUNC_NAME, double)
+
+#define UGENINT_GEQ8BIT_UNARY_CHECK(FUNC_NAME)                                 \
```

Comment:
What does 8bit refer to here?

Reply:
It should be `GEQ16BIT` it refers to the fact that it's =>16bit integer types, but this is a typo.

Reply:
I'd prefer `UGENINT_NAN_UNARY_CHECK`, but can live with the current name.